packages/shared/src/components/post/PostActions.tsx

```diff
@@ -166,14 +172,30 @@ export function PostActions({
             Bookmark
           </QuaternaryButton>
           <QuaternaryButton
-            id="share-post-btn"
+            id="copy-post-btn"
             onClick={() => onCopyLinkClick(post)}
             icon={<LinkIcon />}
             responsiveLabelClass={actionsClassName}
-            className="btn-tertiary-cabbage"
+            className={classNames(
+              'btn-tertiary-cabbage',
+              shareExperience && 'hidden tablet:flex',
+            )}
           >
             Copy
           </QuaternaryButton>
+          {shareExperience && (
+            <QuaternaryButton
+              id="share-post-btn"
+              onClick={() =>
+                globalThis?.navigator?.share
+                  ? openNativeSharePost(post)
+                  : openSharePost({ post })
```

Comment:
Looking at this code, my refactoring urge is growing. `openSharePost` accepts an object, but `openNativeSharePost` accepts an argument instead.

Also, since both `openSharePost` and `openNativeSharePost` come from the `useSharePost` hook, maybe the hook can provide a single function instead and move the `globalThis?.navigator?.share` logic into the hook itself?

Reply:
Good point. Yeah, we can probably abstract this.